### ISSUE TITLE
Resource action template body could not support primitive type

### ISSUE THREAD

**Issue by tadelesh** [Azure/typespec-azure#57]
For brown field RPs, there exist post operations of a resource which body param is basic type. See [example](https://github.com/Azure/azure-rest-api-specs/blob/0077b4a8c5071d3ab33fd9f9ba013581c9a66b8f/specification/alertsmanagement/resource-manager/Microsoft.AlertsManagement/preview/2019-05-05-preview/AlertsManagement.json#L250).
Current arm template could not support it.

**Comment by markcowl** [comment]
@tadelesh  Why couldn't you use [this pattern ](https://azure.github.io/typespec-azure/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3Jl3yvIK3Jlc291cmNlLW1hbmFnZXIiOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TRFVbpAKnIG0HEZS5Db3JlzhJSx3dNxnY7CgovKiogQ29udG9zbyDIHiBQcm92aWRlciDmAJ5tZW50IEFQSS4gKi8KQGFybcggTmFtZXNwYWNlCkBzZXJ2aWNlKHsKICB0aXRsZTogIsdYyC5IdWJDbGllbnQiLAp9KQpA5wFgZWQo5wDBcykKbshSIE1pY3Jvc29mdC7SR%2B8AuEFQSSDHTXPkAKNlbnVtIMhUIOQAksQuMjAyMS0xMC0wMS1wcmV2aWV3yDXENCAgQHVzZURlcGVuZGVuY3ko9QEx6AFrcy52MV8wX1DGSF8xKcRAYXJtQ29tbW9u5AGZc8cq10jLKctUNcRIYPIAqWAsCn3mAPNB6AD16wEOIOgCTeQAyG1vZGVsIEVtcGxveWVlIGlzIFRyYWNrZWToAII8yBxQcm9wZXJ0aWVzPuUBJC4u6QCm5AHYUGFyYW1ldGVyyTE%2BO%2BgAhslfcMlE0nzKYOkBg0FnZSBvZiBlyD%2FlAXhhZ2U%2FOiBpbnQzMjsKxylDaXR50ipjaXR5Pzogc3Ry5QLrxyxQcm9maWzTWUBlbmNvZGUoImJhc2U2NHVybCLkAWBwxjA%2FOiBieXRlc8lIVGhlIHN0YXR1c8RLdGhlIGxhc3Qg5ADFYXRpb27lAv4gIEB2aXNpYmlsaXR5KCJyZWFkyFXEGOUDhVN0YXTEX%2BUBo8wU6QFAxGvMMsV45QDCYekBzcVvQGxyb8Q7dXMKdW7kAqrRVOUBXOYBEizsAMrIRyBjcmXEJ3JlcXVlc3QgaGFzIGJlZW4gYWNjZXB0ZWTEZyAgQccOOiAiyAsi1lBpxEDkALTpAMHIROwAnDogIswP2kx1cGRhdMRPxUNVxw46ICLIC8o76QSf6QDE5gDcZOcBmlN1Y2NlZWTlAMXJDNM%2FxTbkAU1mYWlsyT5GxQ06ICLGCdw4d2FzIGNhbmNlyj5Dxw%2FkBPbHC%2F8BQCBkZWxl6QGARMQN5gD5yAvkBSbvA29tb3bqAcrpA3FNb3ZlUscV6ANqxHNtb3bEaGZyb20gbG9j5gC8xW7EE%2FEDSMszdG%2FPMXRvyi%2F3AJVzcG9uc%2BsEg%2BYAlscW7ACX7gNkxT7FZMZ85gLuzW5pbnRlcmbkBhRP6AOMcyBleHRlbmRz9gbYLsspe30K5Qa0yCPKG8tZ6ADN5gSrZ2V05AGnQco1UmVhZOwFCCDnAntPcuUCp%2BUB1ssvQ8cdUmVwbGFjZUFzeW5jzj%2FlAvfQN1BhdGNoU8wsLPMFTsVA5gI9z0DlAjplV2l0aG91dE9r03dsaXN0QnnIMEdyb3Vwz0RMxSJQYXJlbnTUPFN1YnNjcmlw5QIcxjvGM8wZzDnoBfsgc2FtcGzrAwlhY8VEdGhhdOYCBukFvnRvIGRpZmZl5ACE7wKJxSnuALJBxUjvATfrAxTIDeYCi%2BYAkuQBGsY42k0KICAg6QF%2FLMUO5AIdxAFAYm9keSDEBeoCqcQYfd8nyCcKICDlAVNh%2FwCI5gHC%2FwCJ1WLfJ%2B4AifABo0hFQUTqBvjkAY9jaGVja%2BoBu2V4aXN0ZW7mCHIgxh5FyRTvAwzNHe4ITQ%3D%3D&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-rulesets%2Fresource-manager%22%5D%7D%7D)(see syncAction and asyncAction operations)

**Comment by tadelesh** [comment]
yes, you are right. when this issue open, i don't believe we have clear story for `@body`, but now it is doable. thanks for the solution. let me close this issue. @pshao25 please pay attention to this solution that you need to add to converter.

**Comment by pshao25** [comment]
@markcowl Do you have any concern to make `Request` in `ArmResourceActionSync` accept a primitive type? Your solution looks very like a workaround to me. How could customers know they could write this way if they don't ask us?

**Comment by markcowl** [comment]
This isn't a workaround, it is the same pattern we suggest in Azure.Core. the request could contain headers and the like, and this pattern makes it clear exactly how the request is serialized

**Comment by pshao25** [comment]
That makes sense.
